src/vulkan/engine_vulkan.cc

```diff
@@ -450,6 +450,85 @@ Result EngineVulkan::DoDrawRect(const DrawRectCommand* command) {
   return {};
 }
 
+Result EngineVulkan::DoDrawGrid(const DrawGridCommand* command) {
+  auto& info = pipeline_map_[command->GetPipeline()];
+  if (!info.vk_pipeline->IsGraphics())
+    return Result("Vulkan::DrawGrid for Non-Graphics Pipeline");
+
+  auto* graphics = info.vk_pipeline->AsGraphics();
+
+  float x = command->GetX();
+  float y = command->GetY();
+  float width = command->GetWidth();
+  float height = command->GetHeight();
+  const uint32_t columns = static_cast<uint32_t>(command->GetColumns());
+  const uint32_t rows = static_cast<uint32_t>(command->GetRows());
+  const uint32_t vertices = columns * rows * 3 * 2;
+
+  if (command->IsOrtho()) {
+    const float frame_width = static_cast<float>(graphics->GetWidth());
+    const float frame_height = static_cast<float>(graphics->GetHeight());
+    x = ((x / frame_width) * 2.0f) - 1.0f;
+    y = ((y / frame_height) * 2.0f) - 1.0f;
+    width = (width / frame_width) * 2.0f;
+    height = (height / frame_height) * 2.0f;
+  }
+
+  std::vector<Value> values(vertices * 2);
+
```

Comment:
Lets pull out the cell size calculations:

```
double cell_width = width / columns;
double cell_height = height / rows;
```

Reply:
Done